subsys/bluetooth/scan.c

```diff
@@ -938,6 +1109,25 @@ static void scan_default_param_set(void)
 	bt_scan.scan_param = *scan_param;
 }
 
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	if (IS_ENABLED(CONFIG_BT_SCAN_CONNECTION_ATTEMPTS_FILTER)) {
+		scan_attempts_filter_device_add(bt_conn_get_dst(conn));
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	if (IS_ENABLED(CONFIG_BT_SCAN_CONNECTION_ATTEMPTS_FILTER)) {
```

Comment:
It is recommended to clear this filter after every scanning session.  I assume that for example 3 disconnection from the same device during outgoing scanning session can be treated as error. It would be good to prevent this but I don`t know if it is possible

Reply:
I think that, from the application perspective, we could use the current approach, but we need to state in documentation (or in Kconfig help) that the module counts all the disconnections and the application is responsible to clean the connection attempts filter device list (the required information is already added to the documentation).
For example the nrf_desktop would clean the list after every successful connection (peripheral connected, established security and has all the required GATT characteristics). This approach works fine for us.
I think that main purpose of connection attempts filter is to prevent central from connecting and disconnecting from a peripheral forever in a loop and a few failed connection tries from time to time would not be a problem.

Reply:
Alternatively, the application could use the blocklist and handle adding devices.

Reply:
@MarekPieta Thank you for explaining it. I think it would be good to have this recommended approach documented.